tests/handlers.py

```diff
@@ -0,0 +1,142 @@
+        errors = 0
+        for handler_version in control.commandMap:
+            # test for commands
+            for command_name, command_class in control.commandMap[handler_version].items():
+                command_version = getVersionFromCommandClass(command_class)
+
+                # skip commands with the same version as their handler
+                # b/c they are obviously the latest possible version
+                if command_version == handler_version:
+                    continue
+
+                # load the module defining all possible command implementations
+                command_module = importlib.import_module(command_class.__module__)
+
+                for impl_class in command_module.__dict__.values():
+                    # skip everything which isn't a class
+                    if type(impl_class) is not type:
+                        continue
+
+                    # skip everything which doesn't inherit from KickstartCommand
+                    if not issubclass(impl_class, KickstartCommand):
+                        continue
+
+                    # skip everything which inherits from DeprecatedCommand
+                    # b/c we have multiple inheritance in order to use
+                    # KickstartCommand._getParser() for documentation purposes.
+                    if issubclass(impl_class, DeprecatedCommand):
+                        continue
+
+                    # skip base classes as well
+                    if impl_class.__name__ in ['KickstartCommand', 'DeprecatedCommand']:
+                        continue
+
+                    # version of this particular implementation
+                    impl_version = getVersionFromCommandClass(impl_class)
+
+                    # skip older implementations of the same command
+                    if impl_version <= command_version:
+                        continue
+
+                    # if there is a newer implementation, which isn't newer than the
+                    # handler version this means our commandMap doesn't contain the
+                    # latest possible version for this command. For example we're testing
+                    # the f23.py handler so handler_version == F23, command_version == F20
+                    # and impl_version == F21 or later. This is a bug!
+                    if command_version < impl_version and impl_version <= handler_version:
+                        s_handler_version = versionToString(handler_version)
+                        s_impl_version = versionToString(impl_version)
+                        # skip newer RHEL implementations if testing a Fedora handler
+                        # or vice versa
+                        if (s_handler_version.startswith('F') and \
+                            s_impl_version.startswith('RHEL')) or \
+                            (s_handler_version.startswith('RHEL') and \
+                            s_impl_version.startswith('F')):
+                            continue
```

Comment:
I don't like this block. It introduces the possibility to silent legitimate errors. For example modify `handlers/rhel7.py` to contain "autopart': pykickstart.commands.autopart.F20_AutoPart then run the test. The error isn't detected. 

Part of the problem is the fact that RHEL7_AutoPart = F21_AutoPart. If we define the class like this
```
class RHEL7_AutoPart(F21_AutoPart):
    pass
```

then the error is detected even with the skip block in place. Maybe I can add another test for those Fxy_Command = Fxx_Command definitions and update the code where necessary. What do you guys think ?